psyneulink/core/components/functions/userdefinedfunction.py

```diff
@@ -29,6 +30,295 @@
 
 __all__ = ['UserDefinedFunction']
 
+class UserDefinedFunctionVisitor(ast.NodeVisitor):
```

Comment:
This should be generalized and moved to `codegen.py`. The `udf` parameter is only ever used to get param ids which can be passed explicitly.
UDF can include simple child wrapper that takes udf param and passes param ids to the parent.